diff --git a/libs/androidfw/ApkParsing.cpp b/libs/androidfw/ApkParsing.cpp
index 7eedfdb5..937d7d73 100644
--- a/libs/androidfw/ApkParsing.cpp
+++ b/libs/androidfw/ApkParsing.cpp
@@ -17,6 +17,7 @@
 #include "androidfw/ApkParsing.h"
 #include <algorithm>
 #include <array>
+#include <cstring>
 #include <stdlib.h>
 #include <string_view>
 #include <sys/types.h>
diff --git a/libs/androidfw/AssetsProvider.cpp b/libs/androidfw/AssetsProvider.cpp
index 2d3c0650..9d7119d0 100644
--- a/libs/androidfw/AssetsProvider.cpp
+++ b/libs/androidfw/AssetsProvider.cpp
@@ -104,7 +104,7 @@ std::unique_ptr<ZipAssetsProvider> ZipAssetsProvider::Create(std::string path,
     return {};
   }
 
-  struct stat sb{.st_mtime = -1};
+  struct stat sb{}; sb.st_mtime = -1;
   // Skip all up-to-date checks if the file won't ever change.
   if (!isReadonlyFilesystem(path.c_str())) {
     if ((released_fd < 0 ? stat(path.c_str(), &sb) : fstat(released_fd, &sb)) < 0) {
@@ -137,7 +137,7 @@ std::unique_ptr<ZipAssetsProvider> ZipAssetsProvider::Create(base::unique_fd fd,
     return {};
   }
 
-  struct stat sb{.st_mtime = -1};
+  struct stat sb{}; sb.st_mtime = -1;
   // Skip all up-to-date checks if the file won't ever change.
   if (!isReadonlyFilesystem(released_fd)) {
     if (fstat(released_fd, &sb) < 0) {
diff --git a/libs/androidfw/BigBufferStream.cpp b/libs/androidfw/BigBufferStream.cpp
index f18199cf..0c3ca8a8 100644
--- a/libs/androidfw/BigBufferStream.cpp
+++ b/libs/androidfw/BigBufferStream.cpp
@@ -17,6 +17,7 @@
 #include "androidfw/BigBufferStream.h"
 
 #include <algorithm>
+#include <limits>
 
 namespace android {
 
diff --git a/libs/androidfw/PngCrunch.cpp b/libs/androidfw/PngCrunch.cpp
index cf3c0eef..29875e5a 100644
--- a/libs/androidfw/PngCrunch.cpp
+++ b/libs/androidfw/PngCrunch.cpp
@@ -18,6 +18,7 @@
 #include <zlib.h>
 
 #include <algorithm>
+#include <limits>
 #include <unordered_map>
 #include <unordered_set>
 
diff --git a/tools/aapt2/ApkInfo.proto b/tools/aapt2/ApkInfo.proto
index b5ff71fa..044cca35 100644
--- a/tools/aapt2/ApkInfo.proto
+++ b/tools/aapt2/ApkInfo.proto
@@ -16,7 +16,7 @@
 
 syntax = "proto3";
 
-import "frameworks/base/tools/aapt2/Resources.proto";
+import "Resources.proto";
 
 package aapt.pb;
 
diff --git a/tools/aapt2/AppInfo.h b/tools/aapt2/AppInfo.h
index cabbe7ea..13065572 100644
--- a/tools/aapt2/AppInfo.h
+++ b/tools/aapt2/AppInfo.h
@@ -17,6 +17,7 @@
 #ifndef AAPT_APP_INFO_H
 #define AAPT_APP_INFO_H
 
+#include <cstdint>
 #include <optional>
 #include <set>
 #include <string>
diff --git a/tools/aapt2/Resources.proto b/tools/aapt2/Resources.proto
index 1d7fd1d1..e6861d4e 100644
--- a/tools/aapt2/Resources.proto
+++ b/tools/aapt2/Resources.proto
@@ -16,7 +16,7 @@
 
 syntax = "proto3";
 
-import "frameworks/base/tools/aapt2/Configuration.proto";
+import "Configuration.proto";
 
 package aapt.pb;
 
diff --git a/tools/aapt2/ResourcesInternal.proto b/tools/aapt2/ResourcesInternal.proto
index b0ed3da3..97aa5af7 100644
--- a/tools/aapt2/ResourcesInternal.proto
+++ b/tools/aapt2/ResourcesInternal.proto
@@ -16,8 +16,8 @@
 
 syntax = "proto3";
 
-import "frameworks/base/tools/aapt2/Configuration.proto";
-import "frameworks/base/tools/aapt2/Resources.proto";
+import "Configuration.proto";
+import "Resources.proto";
 
 package aapt.pb.internal;
 
diff --git a/tools/aapt2/cmd/Command.h b/tools/aapt2/cmd/Command.h
index 1416e980..8c3ec9d1 100644
--- a/tools/aapt2/cmd/Command.h
+++ b/tools/aapt2/cmd/Command.h
@@ -18,6 +18,7 @@
 #define AAPT_COMMAND_H
 
 #include <functional>
+#include <memory>
 #include <optional>
 #include <ostream>
 #include <string>
diff --git a/tools/aapt2/io/StringStream.cpp b/tools/aapt2/io/StringStream.cpp
index bb3911b2..c399fe5d 100644
--- a/tools/aapt2/io/StringStream.cpp
+++ b/tools/aapt2/io/StringStream.cpp
@@ -16,6 +16,9 @@
 
 #include "io/StringStream.h"
 
+#include <cstring>
+#include <limits>
+
 using ::android::StringPiece;
 
 namespace aapt {
